src/interfaces.d.ts

```diff
@@ -35,18 +44,18 @@ export interface HasSortEvent {
 }
 
 export interface HasItem {
-	item: ItemProperties<any>;
+	item: ItemProperties;
 }
 
 export interface HasItems {
-	items: ItemProperties<any>[];
+	items: ItemProperties[];
 }
 
 export interface HasValue {
 	value: string;
 }
 
-export interface ItemProperties<T> {
+export interface ItemProperties<T = any> {
```

Comment:
This doesn't really have anything to do with the data - it's what the data provider gives to the grid. `ArrayDataProvider` still supports simple data with `idProperty`.